datafusion/substrait/tests/utils.rs

```diff
@@ -0,0 +1,154 @@
+#[cfg(test)]
+pub(crate) mod test {
+    use datafusion::catalog_common::TableReference;
+    use datafusion::datasource::empty::EmptyTable;
+    use datafusion::prelude::SessionContext;
+    use datafusion_substrait::extensions::Extensions;
+    use datafusion_substrait::logical_plan::consumer::from_substrait_named_struct;
+    use std::fs::File;
+    use std::io::BufReader;
+    use std::sync::Arc;
+    use substrait::proto::read_rel::{NamedTable, ReadType};
+    use substrait::proto::rel::RelType;
+    use substrait::proto::{Plan, ReadRel, Rel};
+
+    pub(crate) fn read_json(path: &str) -> Plan {
+        serde_json::from_reader::<_, Plan>(BufReader::new(
+            File::open(path).expect("file not found"),
+        ))
+        .expect("failed to parse json")
+    }
+
+    pub(crate) struct TestSchemaCollector {
+        ctx: SessionContext,
+    }
```

Comment:
This collector is based on a similar bit of tooling in [Isthmus](https://github.com/substrait-io/substrait-java/blob/d6256487f75032e2e8fc803976d916e4a5bc44bd/isthmus/src/main/java/io/substrait/isthmus/SubstraitToCalcite.java#L102-L128), a Substrait library used for integrating with Apache Calcite.